OpenRA.Mods.Common/Traits/Cloak.cs

```diff
@@ -17,6 +17,20 @@
 
 namespace OpenRA.Mods.Common.Traits
 {
+	[Flags]
+	public enum UncloakType
+	{
+		None = 0,
+		Attack = 1,
+		Move = 2,
+		Unload = 4,
+		Infiltrate = 8,
+		Demolish = 16,
+		Damage = 32,
+		[Desc("Uncloaks on Attack, Unload, Infiltrate, Demolish")]
```

Comment:
The problem I see arising when maintaining a mod is that most of the time you will want default behavior + your additional uncloak types. If that list gets extended in the future and you have to explicitely state all the default types, you will have to carefully adapt all those units using Cloak.
Anyways, removed that pseudo flag and rebased